encoding/da.go

```diff
@@ -465,6 +465,52 @@ func checkCompressedDataCompatibility(data []byte) error {
 	return nil
 }
 
+// Fast testing if the compressed data (v7) is compatible with our circuit
+// (require specified frame header and each block is compressed)
+func checkCompressedDataCompatibilityV7(data []byte) error {
```

Comment:
what exactly is the difference between this method and `checkCompressedDataCompatibility`?

Reply:
the only difference is this check is removed:
```
		if blkType != 2 {
			return fmt.Errorf("unexpected blk type {%d}, size {%d}, last {%t}", blkType, blkSize, isLast)
		}
```
want to keep the behavior consistent for previous codec versions, so adding a new function, but it's open to change (might also affect some unit test results of previous codec versions).